---
title: Lexicon
author: Sylvain SCHMITT
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    theme: journal
    toc: yes
    toc_depth: 5
    toc_float: yes
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

# Amplified Fragment Length Polymorphism (AFLP)

Is a PCR-based tool used in genetics research, DNA fingerprinting, and in the practice of genetic engineering. Developed in the early 1990s by Keygene, AFLP uses restriction enzymes to digest genomic DNA, followed by ligation of adaptors to the sticky ends of the restriction fragments. A subset of the restriction fragments is then selected to be amplified. This selection is achieved by using primers complementary to the adaptor sequence, the restriction site sequence and a few nucleotides inside the restriction site fragments. The amplified fragments are separated and visualized on denaturing polyacrylamide gels, either through autoradiography or fluorescence methodologies, or via automated capillary sequencing instruments.

# Chloroplast DNS (cpDNA)

Just for the abbreviation.

# double digest RAD sequencing (ddRAD-Seq)

Just for abbreviation, see Restriction site Associated DNA for RAD definition.

# Genotyping-by-Sequencing (GbS)

In the field of genetic sequencing, genotyping by sequencing, also called GBS, is a method to discover single nucleotide polymorphisms (SNP) in order to perform genotyping studies, such as genome-wide association studies (GWAS). GBS uses restriction enzymes to reduce genome complexity and genotype multiple DNA samples. After digestion, PCR is performed to increase fragments pool and then GBS libraries are sequenced using next generation sequencing technologies, usually resulting in about 100bp single-end reads. It is relatively inexpensive and has been used in plant breeding. Although GBS presents an approach similar to restriction-site-associated DNA sequencing (RAD-seq) method, they differ in some substantial ways.

# Internal Transcribed Spacer (ITS)

Refers to the spacer DNA situated between the small-subunit ribosomal RNA (rRNA) and large-subunit rRNA genes in the chromosome or the corresponding transcribed region in the polycistronic rRNA precursor transcript. Sequence comparison of the ITS region is widely used in taxonomy and molecular phylogeny because of several favorable properties:

* it is routinely amplified thanks to its small size associated to the availability of highly conserved flanking sequences ;
* it is easy to detect even from small quantities of DNA due to the high copy number of the rRNA clusters ;
* it undergoes rapid concerted evolution via unequal crossing-over and gene conversion. This promotes intra-genomic homogeneity of the repeat units, although high-throughput sequencing showed the occurrence of frequent variations within plant species.
* it has a high degree of variation even between closely related species. This can be explained by the relatively low evolutionary pressure acting on such non-coding spacer sequences.

# Restriction site associated DNA markers (RAD)

Restriction site associated DNA (RAD) markers are a type of genetic marker which are useful for association mapping, QTL-mapping, population genetics, ecological genetics and evolution. The use of RAD markers for genetic mapping is often called RAD mapping. An important aspect of RAD markers and mapping is the process of isolating RAD tags, which are the DNA sequences that immediately flank each instance of a particular restriction site of a restriction enzyme throughout the genome. Once RAD tags have been isolated, they can be used to identify and genotype DNA sequence polymorphisms mainly in form of single nucleotide polymorphisms (SNPs). Polymorphisms that are identified and genotyped by isolating and analyzing RAD tags are referred to as RAD markers.

# Single-Nucleotide Polymorphism (SNP) 

Variation in a single nucleotide that occurs at a specific position in the genome, where each variation is present to some appreciable degree within a population (e.g. > 1%).

# Simple Sequence Repeats (SSR) 

A microsatellite is a tract of repetitive DNA in which certain DNA motifs (ranging in length from 2–13 base pairs) are repeated, typically 5–50 times. Microsatellites occur at thousands of locations within an organism's genome; additionally, they have a higher mutation rate than other areas of DNA leading to high genetic diversity. Nuclear microsatellite are called nSSR.

# References