---
title: Symphonia preliminary anlaysis
author: Sylvain SCHMITT
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    theme: journal
    toc: yes
    toc_depth: 5
    toc_float: yes
csl: /home/sylvain/Documents/Bibliography/csl/mee.csl
bibliography: /home/sylvain/Documents/Bibliography/library.bib
link-citations: yes
---

```{r set, message=FALSE, warning=FALSE, echo=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
mpath <- "~/Documents/BIOGECO/PhD/documents/preliminary analysis/preliminary_models/"
library(knitr)
library(readr)
library(dplyr)
library(tidyr)
library(ggplot2)
library(RColorBrewer)
library(reshape2)
library(rstan)
library(bayesplot)
library(GGally)
library(vcfR)
library(poppr)
library(ape)
library(starmie)
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
    cache = T, cache.lazy = F)
```

# Introduction

View [PhD coneptual document online](https://docs.google.com/document/d/1ws7OHgMXHwdHyw5Se_nb_xQk1eDelew8lF4viN7wCpQ/edit?usp=sharing).

# Data preparation

## Genetic data

### Assembly

We gathered genetic material (ddRADseq) of french Guiana from Torroba-Balmori unpublished data (Paracou and Regina). We cleaned `fastq` files after a check with `fastQCheck` allowing us to correct two sequences by removing theim for individuals $PR_{49}$ and $RG_1$. We used `ipyrad ` for the interactive assembly of ddRADseq data sets on genotoul cluster (with denovo assembly, AATT and AT restriction overhang, 85% clustering threshold and a minimum of 48 sample per locus).

### Data preparation

```{r data prep}
# opening
# path <- "~/Documents/BIOGECO/PhD/data/GbS_Symphonia/assembly/symphoGbS2_outfiles/"
# data <- read.vcfR(file.path(path, 'symphoGbS2.vcf'), verbose = F)
# data <- vcfR2genlight(data)
# ploidy(data) <- 2
# save(data, file = './preliminary_save/assembly.Rdata')
load('./preliminary_save/assembly.Rdata')

# population data
ID0 <- data$ind.names
data$ind.names[which(data$ind.names %in% c('PR_49c', 'RG_1c'))] <- c('PR_49', 'RG_1')
path <- "~/Documents/BIOGECO/PhD/data/links/"
pop <- data.frame(ID0 = ID0, pop_num = data$ind.names) %>% 
  left_join(read_csv(file.path(path, 'Symphonia_ind_names.csv'))) %>% 
  left_join(read_csv(file.path(path, 'Symphonia_ind_coords.csv'),
                     col_types = cols(Latitude = col_character(), 
        Longitude = col_character())),
            by = c("ID" = "Ind_ID")) %>% 
  select(ID0, pop_num, Pop, Species, Latitude, Longitude) %>%
  separate(Species, c("Genus", "species"), sep = " ") %>%
  mutate(species = substr(species, 0,2)) %>%
  mutate(Pop = paste0(Pop, '_', species)) %>%
  select(ID0,pop_num, Pop, Latitude, Longitude) %>%
  rename(ID = pop_num) %>% 
  mutate(Latitude = as.numeric(gsub(',', '.', Latitude)),
         Longitude = as.numeric(gsub(',', '.', Longitude)))
rm(ID0)
pop(data) <- pop$Pop

# preparing data corrdinates for SPAGEDI
pop[which(pop$ID == 'PR_38'),]$Longitude <- -pop[which(pop$ID == 'PR_38'),]$Longitude
## TODO: transform data coordinates to UTM
other(data) <- list(Longitude = pop$Longitude,
                    Latitude = pop$Latitude)

# Writting pop data table for vcf2str
pop %>% 
  select(ID0, Pop) %>% 
  arrange(Pop) %>% 
  write.table('./preliminary_save/pop.txt',
              quote = F, col.names = F, row.names = F)
```

We used `vcfR` to load SNPs data into R, and we transform it in genligh object for `adegenet`. We related indivdual IDs to their population and coordinates with *links* table. We coded population in 4 subset for *Symphonia globulifera* and *sp1* in both Paracou and Régina ($PR_{gl}$, $PR_{sp}$, $RG_{gl}$, $RG_{sp}$). Population definition was used to transform vcf file to structure file with PGDspider for further genetic structure analysis with STRUCTURE software. We corrected and transformed in UTM coordinates to compute kinship distance matrix with SPAGEDI.

### Population structure

#### Within R

We used `poppr` package to compute population tree with bitwise distance.

```{r tree}
# tree <- aboot(data, tree = "upgma", dist = bitwise.dist, sample = 100, cutoff = 50, showtree = F, quiet = T)
# save(tree, file = './preliminary_save/tree.Rdata')
load('./preliminary_save/tree.Rdata')
cols <- brewer.pal(n = nPop(data), name = "Dark2")
plot.phylo(tree, cex = 0.8, font = 2, adj = 0, tip.color =  cols[pop(data)])
legend('topleft', legend = c("Paracou globulifera", "Paracou sp1", 
                             "Régina globulifera", "Régina sp1"), fill = cols, border = FALSE, bty = "n", cex = 2)
axis(side = 1)
title(xlab = "Genetic distance (proportion of loci that are different)")
rm(tree)
```

We do not observed disctinct cluster among the tree.

We thus used a principal components analysis (PCA) converts the observed SNP data into a set of values of linearly uncorrelated variables with `adegenet` package.

```{r pca}
# save(data, file = './preliminary_save/pcadata.Rdata')
# load('./preliminary_save/pcadata.Rdata')
# data.pca <- glPca(data, nf = 3)
# save(data.pca, file = './preliminary_save/glpca.Rdata')
load('./preliminary_save/glpca.Rdata')
data.pca.scores <- as.data.frame(data.pca$scores)
data.pca.scores$pop <- pop(data)
ggplot(data.pca.scores, aes(x=PC1, y=PC2, colour=pop)) + 
  geom_point(size=2) + stat_ellipse(level = 0.95, size = 1) +
  scale_color_manual(values = cols) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) + theme_bw()
rm(data.pca.scores)
```

Populations, as expected, are clustered by Species and not location on the PCA first and second axis plan.

We then performed disciminant analysis on the PCA results and used a composite stacked bar plot to visualize the posterior assignments of each sample.

```{r compoplot}
data.dapc <- dapc(data, n.pca = 3, n.da = 2, glPca = data.pca)
dapc.results <- as.data.frame(data.dapc$posterior)
dapc.results$pop <- pop(data)
dapc.results$indNames <- rownames(dapc.results)
dapc.results <- melt(dapc.results)
colnames(dapc.results) <- c("Original_Pop","Sample","Assigned_Pop","Posterior_membership_probability")
ggplot(dapc.results, aes(x=Sample, y=Posterior_membership_probability, fill=Assigned_Pop)) + geom_bar(stat='identity') + 
  scale_fill_manual(values = cols) + 
  facet_grid(~Original_Pop, scales = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))
rm(data.dapc, dapc.results, data.pca, col, pop)
```

We observed ...

#### Within STRUCTURE

We examined gene pool structure with STRUCTURE software to exclude any sampling mistakes or early-generation hybrids. **Operation running**.

```{r structure}
# test <- loadStructure('~/Documents/BIOGECO/PhD/data/GbS_Symphonia/structure/results_f/k3/symphoGbS2_k3_run1_f')
```


### Disctance matrix

**We are planning to compute kniship matrix from SNPs identified in previous assembly with SPAGEDI to be used in further models.**

```{r spagedi prep}
# path <- "~/Documents/BIOGECO/PhD/data/GbS_Symphonia/assembly/symphoGbS2_outfiles/"
# data <- read_delim(file.path(path, "symphoGbS2.str"), " ", escape_double = FALSE, trim_ws = TRUE)
# data <- data %>% 
#   separate("\t\tSNP_1", c("Ind", "Pop", "SNP_1"), sep = "\t") %>% 
#   mutate(SNP_1 = as.integer(SNP_1)) %>% 
#   group_by(Ind, Pop) %>% 
#   mutate_all(funs(replace(., .==-9, 0))) %>% 
#   mutate_all(funs(paste0(., collapse = ""))) %>% 
#   unique()
# data <- pop %>% 
#   select(ID0, Pop, Latitude, Longitude) %>% 
#   left_join(data, by = c("ID0" = "Ind")) %>% 
#   select(-Pop.y)
# names(data)[1:4] <- c('Ind', 'Pop', 'Xutm', 'Yutm')
# names(data)[-c(1:4)] <- gsub('SNP_', '', names(data)[-c(1:4)])
# fileConn <- file(file.path(path, "symphoGbS2.spag"))
# writeLines(c('80\t4\t2\t3098\t1\t2', '-7'), fileConn)
# close(fileConn)
# suppressWarnings(write.table(data, file.path(path, "symphoGbS2.spag"),
#                              sep = "\t", append = TRUE,
#                              row.names = FALSE, quote = FALSE))
# cat("END", file = file.path(path, "symphoGbS2.spag"), append = TRUE)
# rm(data, fileConn)
path <- "~/Documents/BIOGECO/PhD/data/GbS_Symphonia/distance_matrices"
data <- rSpagedi::makeSpagediList(path_to_out = file.path(path, "symphoGbS2_kniship.txt"))
data$kin
```


## Growth data

```{r Paracou data}
# sp <- read_delim(file.path(path, 'Paracou_species.csv'), ',') %>%
#   filter(Genre == "Symphonia") %>%
#   rename(idSp = idTaxon, Family = Famille, Genus = Genre, species = espece) %>%
#   dplyr::select(-Family)
# data <- rbind(read_delim(file.path(path, 'R_PreExploit_1984-85-86.csv'), ';'),
#            read_delim(file.path(path, 'R_PostExploit_1987-88.csv'), ';'),
#            read_delim(file.path(path, 'R_Paracou_1988-2016.csv'), ';')) %>%
#   filter(code_vivant == 'VRAI') %>%
#   filter(idTaxon %in% sp$idSp) %>%
#   dplyr::select(n_parcelle, n_carre, n_arbre, idArbre, campagne, Xutm, Yutm, circonf, idTaxon) %>%
#   mutate(circonf = circonf*10 / pi) %>%
#   rename(plot = n_parcelle, square = n_carre, id = n_arbre, idPR = idArbre,
#          census = campagne, X = Xutm, Y = Yutm, dbh = circonf, idSp = idTaxon) %>% 
#   filter(!is.na(dbh))
# rm(sp)
# save(data, file = './preliminary_save/data.Rdata')
load('./preliminary_save/data.Rdata')
```

We gathered data from Paracou between 1984 and 2016. We kept tree only flagged as alive, with a measured dbh and with species ID corresponding to *Symphonia* genus. We thus obtained 27868 entries.

**We need to better clean Paracou diameter census before subsequent anlaysis !**

```{r growth data}
data <- data %>% 
  select(idPR, census, dbh, idSp) %>% 
  group_by(idPR) %>% 
  arrange(idPR, census) %>%
  mutate(ddbh = (dbh - lag(dbh, default=first(dbh)))) %>% 
  mutate(dt = (census - lag(census, default=first(census)))) %>% 
  filter(ddbh > 0) %>%
  filter(dt != 0) %>%
  filter(dt < 3) %>%
  mutate(AGR = ddbh/dt) 
  # filter(AGR < 10)
# data <- data[sample.int(dim(data)[1], 1000),] # subset for tests
data %>%
  ggplot(aes(dbh, AGR)) +
  geom_point(alpha = 0.3) +
  xlab('Diameter at breast height (dbh in mm)') +
  ylab(expression('Annual growth rate (AGR in'~mm.year^{-1}~")")) +
  ylim(0, 10)
```

We computed the tree annual growth ($AGR$, in $mm.year^{-1}$) as the difference of diameter at breast height ($dbh$, in $mm$) divided by the time step (from 1 to 2 year depending on the censuses). 

# Modelling

## Model M0

We related $AGR$ to diameter at breast height with the Canham model following @herault_functional_2011 :
$$log(AGR_i) = max * exp(-\frac{1}{2}*log(\frac{dbh_i}{\frac{dopt}{ks}})^2)$$
Where $max$ represents the maximum value of $log(AGR_i +1)$, $dopt$ the optimal $dbh$ when the maximum is reached and $ks$ the kurtosis of the hump shaped distribution.

```{r M0}
# fit <- stan(file = file.path(mpath, 'M0.stan'),
#             data = list(N = dim(data)[1],
#                         dbh = data$dbh,
#                         AGR = data$AGR),
#             chains = 1)
# save(fit, file = file.path(mpath, 'M0.Rdata'))
load(file.path(mpath, 'M0.Rdata'))
```

**Data are not ready, consequently the model is of no interest for the moment. Following graphs are just here for me to have a working code ready for further analysis.**

```{r M0 diagnostic}
# shinystan::launch_shinystan(fit) # interface to vizualise results
mcmc_areas(as.matrix(fit),  prob = 0.8)
mcmc_trace(as.matrix(fit))
ggpairs(data.frame(as.matrix(fit)))
```

# References
